storage/remote/intern.go

```diff
@@ -61,13 +61,6 @@ func (p *pool) intern(s string) string {
 		return ""
 	}
 
-	p.mtx.RLock()
```

Comment:
Rather than removing the optimization to allow concurrent calls to `intern`, all that needs to be done to fix this is to not call `p.mtx.RUnlock()` until after `interned.refs.Inc()`. Though both branches would then need the unlock line.